core/moderation.go

```diff
@@ -218,3 +218,35 @@ func (n *OpenBazaarNode) SetModeratorsOnListings(moderators []string) error {
 	}
 	return n.UpdateIndexHashes(hashes)
 }
+
+func (n *OpenBazaarNode) NotifyModerators(moderators []string) error {
+	settings, err := n.Datastore.Settings().Get()
+	if err != nil {
+		return err
+	}
+	currentMods := make(map[string]bool)
+	if settings.StoreModerators != nil {
+		for _, mod := range *settings.StoreModerators {
+			currentMods[mod] = true
+		}
+	}
+	var addedMods []string
+	for _, mod := range moderators {
+		if !currentMods[mod] {
+			addedMods = append(addedMods, mod)
+			delete(currentMods, mod)
```

Comment:
@cpacia Perhaps I'm thinking incorrectly, but shouldn't this `delete` call be outside the conditional? We want to end up with `currentMods` being a slice of mods not present in `moderators` correct?